### PR TITLE
Markdown Redux Refactor with Tests

### PR DESCRIPTION
**Pull request by DannySuarez**
Test are passing. Refactored code. However, markdown presents the same error as being undefined. Please advise/point us in the right direction.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import Document from './markdown/Document';
 
+
```
Comment (mehtaphysical) [inline src/components/App.js:3]: What is this line for?

File: src/containers/markdownPreview.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { connect } from 'react-redux';
```
Comment (mehtaphysical) [inline src/containers/markdownPreview.js]: Where do you use this file?

Also, the file should be title case

File: src/components/App.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import Document from './markdown/Document';
```
Comment (mehtaphysical) [inline src/components/App.js]: Shouldn't you use your container here?

File: src/components/Markdown.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
```
Comment (mehtaphysical) [inline src/components/Markdown.js]: Why isn't this in the markdown folder?

File: src/containers/markdownPreview.js (hunk 2 above)
Comment (lacrivella) [reply src/containers/markdownPreview.js]: We use it to pass state to Document. Document gets rendered in App. Document is currently our container component. Can we get some more clarification?

Comment (mehtaphysical) [reply src/containers/markdownPreview.js]: I don't think you're using it. You would have to import this file somewhere

File: src/components/markdown/Editor.js [hunk 5]
```diff
@@ -9,7 +9,7 @@ function Editor({ markdown, updateMarkdown }) {
 }
 
 Editor.propTypes = {
-  markdown: PropTypes.string.isRequired,
+  markdown: PropTypes.object.isRequired,
```
Comment (mehtaphysical) [inline src/components/markdown/Editor.js]: Why is this an object instead of a string?

File: src/containers/markdownPreview.js (hunk 2 above)
Comment (mehtaphysical) [reply src/containers/markdownPreview.js]: This file needs to be MarkdownPreview.js

**Review by mehtaphysical — APPROVED** [review]

**Pull request merged**